### PR TITLE
Update the Ollama model hosting details with more clarity

### PR DESCRIPTION
**Pull request by hickeyma**
Added information on installing and running the Ollama server. The information is necessary for users when using Ollama in a notebook.

### CODE REVIEW

**Comment by fayvor** [comment]
Thanks, @hickeyma. Could you instead link to the [Ollama Getting Started instructions](https://github.com/ibm-granite-community/utils/blob/main/recipes/Getting_Started/Getting_Started_with_Ollama.ipynb) and make sure those are complete?

File: recipes/Components/Langchain_LLMs.ipynb [hunk 1]
```diff
@@ -48,7 +48,21 @@
...
+    "\n",
+    "1. [Download and install Ollama](https://ollama.com/download), if you haven't already.\n",
+    "1. Start the Ollama server: `ollama serve`\n",
+    "\n",
+    "#### Install Ollama Server in the Notebook\n",
+    "\n",
+    "Following \"Option B: Shell Script (Linux)\" in [Getting Started with Ollama](https://github.com/ibm-granite-community/utils/blob/main/recipes/Getting_Started/Getting_Started_with_Ollama.ipynb), run the following commands in separate notebook cells:\n",
+    "\n",
+    "1. Install Ollama\n",
+    "2. Start the Ollama server\n",
```
Comment (bjhargrave) [inline recipes/Components/Langchain_LLMs.ipynb:63]: I think you need step 3: `!ollama pull granite-code:3b` to download a model.

**Comment by hickeyma** [comment]
> These instructions are missing the proper pip install command for the langchain_ollama package.

@bjhargrave Could you describe more what is missing? Sorry not sure what you are asking.

**Comment by bjhargrave** [comment]
> Could you describe more what is missing? Sorry not sure what you are asking.

Screenshot 2024-10-01 at 15 06 26

The langchain_ollama package is not present. No prior step in the notebook pip installs the necessary packages. Without a step to install the package, the `from` statement fails.

**Comment by rawkintrevo** [comment]
Iirc, the recipes in this repo are being moved to `snacks` or `kitchen`. @fayvor or @adampingel can comment more.

File: recipes/Getting_Started/Getting_Started_with_Ollama.ipynb [hunk 2]
```diff
@@ -91,6 +93,15 @@
...
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!curl https://ollama.ai/install.sh | sh"
+   ]
```
Comment (fayvor) [inline recipes/Getting_Started/Getting_Started_with_Ollama.ipynb:103]: I don't think we need this again, do we?

File: recipes/Getting_Started/Getting_Started_with_Ollama.ipynb [hunk 3]
```diff
@@ -81,7 +81,9 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "!ollama pull granite-code:3b"
+    "!ollama pull granite-code:3b\n",
+    "!ollama pull granite-code:8b\n",
+    "!ollama pull granite-code:20b"
```
Comment (fayvor) [inline recipes/Getting_Started/Getting_Started_with_Ollama.ipynb:86]: Agreed, the user might as well download these three. Should we also include 34B?

File: recipes/Getting_Started/Getting_Started_with_Ollama.ipynb [hunk 4]
```diff
@@ -64,15 +72,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Pull the model\n",
-    "\n",
-    "IBM Granite Code models available on Ollama can be found [here](https://ollama.com/library/granite-code).\n",
```
Comment (fayvor) [inline recipes/Getting_Started/Getting_Started_with_Ollama.ipynb:69]: Could we keep this link? New models will arrive on Ollama.

File: recipes/Getting_Started/Getting_Started_with_Ollama.ipynb [hunk 5]
```diff
@@ -18,18 +18,28 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Install Ollama\n",
+    "The Ollama server can be run locally on your system or in the notebook itself.\n",
```
Comment (fayvor) [inline recipes/Getting_Started/Getting_Started_with_Ollama.ipynb:21]: Both methods run the server locally (on the same host as the notebook). Maybe "from a terminal" or "from the notebook"?

File: recipes/Getting_Started/Getting_Started_with_Ollama.ipynb [hunk 6]
```diff
@@ -18,18 +18,28 @@
...
-    "[Download from Ollama.com](https://ollama.com/download) and install "
+    "1. [Download and install Ollama](https://ollama.com/download), if you haven't already.\n",
+    "1. Start the Ollama server: `ollama serve`\n",
+    "1. Pull down Granite models:\n",
+    "\n",
+    "    ```shell\n",
+    "    ollama pull granite-code:3b\n",
+    "    ollama pull granite-code:8b\n",
+    "    ollama pull granite-code:20b\n",
+    "    ```"
```
Comment (fayvor) [inline recipes/Getting_Started/Getting_Started_with_Ollama.ipynb:33]: I like the refactoring you've done here: putting the full instructions under each option.

**Review by fayvor — CHANGES_REQUESTED** [review]

**Comment by fayvor** [comment]
> The langchain_ollama package is not present. No prior step in the notebook pip installs the necessary packages.

@bjhargrave I see that. I can fix it when I move these recipes to the Kitchen, if it hasn't been fixed here.

**Comment by fayvor** [comment]
@hickeyma I've moved these changes to a [new PR](https://github.com/ibm-granite-community/granite-kitchen/pull/4) against `granite-kitchen`, where these recipes are migrating. Please review there.

**Comment by fayvor** [comment]
See [this PR](https://github.com/ibm-granite-community/granite-kitchen/pull/4).

**Pull request closed without merging**